### ISSUE TITLE
Feature Selection

### ISSUE THREAD

**Issue by sidhu1012** [Dacience/DataScientist#30]
Implement varoius techniques for feature selection.

**Comment by sidhu1012** [comment]
This is a big issue and should be resolved in multiple PRs

**Comment by parva-jain** [comment]
Can i get assigned for this issue/task?

**Comment by sidhu1012** [comment]
> Can i get assigned for this issue/task?

Yes sure, but this issue is difficult one , so kindly tell me how you are planning to proceed.

**Comment by parva-jain** [comment]
https://towardsdatascience.com/the-5-feature-selection-algorithms-every-data-scientist-need-to-know-3a6b566efd2
I'm planning to implement a few techniques mentioned in this article.
Should I implement one of these and open PR for your review?
Do let me know if you have any suggestions.

**Comment by sidhu1012** [comment]
> https://towardsdatascience.com/the-5-feature-selection-algorithms-every-data-scientist-need-to-know-3a6b566efd2
> I'm planning to implement a few techniques mentioned in this article.
> Should I implement one of these and open PR for your review?
> Do let me know if you have any suggestions.

Go for it!

**Comment by parva-jain** [comment]
So I should create one folder with name feature selection and create a python script in it performing the feature selection task, right?

**Comment by parva-jain** [comment]
> So I should create one folder with name feature selection and create a python script in it performing the feature selection task, right?

and also throw some light on testing or creating an example of any particular technique.

**Comment by sidhu1012** [comment]
> So I should create one folder with name feature selection and create a python script in it performing the feature selection task, right?

Yes.

> > So I should create one folder with name feature selection and create a python script in it performing the feature selection task, right?
> 
> and also throw some light on testing or creating an example of any particular technique.

Design a script, and make test function for the same which checks all possible cases of your script.

**Comment by parva-jain** [comment]
I'm ready with the script of one of the feature selection techniques along with the test script but when I import the main script in the test script in the same way done in the other scripts(ex - test_mae), I get an error - ModuleNotFoundError: No module named 'datascientist'. Even when I executed the test_mae.py script, it threw the same error. Please help me in this particular area. Thanks

**Comment by sidhu1012** [comment]
Have u installed Datascientist and required libraries?

**Comment by sidhu1012** [comment]
Use these commands in terminal

`pip install -e .`
`pip install -r requirements.txt`

**Comment by parva-jain** [comment]
> Have u installed Datascientist and required libraries?

obviously yes...

**Comment by sidhu1012** [comment]
> Use these commands in terminal
> 
> `pip install -e .`
> `pip install -r requirements.txt`

    Then it shouldn't throw that error, please debug and if you haven't used those commands please do.

**Comment by sidhu1012** [comment]
> > Have u installed Datascientist and required libraries?
> 
> obviously yes...
> 

Make sure to install DataSceintist for development.

**Comment by parva-jain** [comment]
If we compare this package with sklearn , there are additional things that need to be added in __init__.py and according to me, that's causing the issue.

**Comment by parva-jain** [comment]
> DataSceintist



> > > Have u installed Datascientist and required libraries?
> > 
> > 
> > obviously yes...
> 
> Make sure to install DataSceintist for development.

By this u mean cloning this repo, right?

**Comment by sidhu1012** [comment]
> > DataSceintist
> 
> 
> 
> > > > Have u installed Datascientist and required libraries?
> > > 
> > > 
> > > obviously yes...
> > 
> > Make sure to install DataSceintist for development.
> 
> By this u mean cloning this repo, right?
> 

After cloning, in the cloned folder you need to create virtual environment, and in that virtual environment you need to run those 2 commands . Then you shoud start development and make PR's.

**Comment by shreyaspj20** [comment]
Can I work on wrapper based methods?

**Comment by sidhu1012** [comment]
> Can I work on wrapper based methods? 

Can you please explain what you have in mind and how would they be beneficial?

**Comment by shreyaspj20** [comment]
Since there are no wrapper based methods like RFE implemented in the project  till now, I feel that this way of feature selection is much more robust than filter based selection based on the fact that RFE also considers the ML algorithm estimator used as one of the hyperparameter while selecting features unlike filter based models.

**Comment by sidhu1012** [comment]
> Since there are no wrapper based methods like RFE implemented in the project  till now, I feel that this way of feature selection is much more robust than filter based selection based on the fact that RFE also considers the ML algorithm estimator used as one of the hyperparameter while selecting features unlike filter based models.

Yes go for it then!

**Comment by parva-jain** [comment]
Can I use sklearn library for implementing chi-square feature selection?

**Comment by sidhu1012** [comment]
> Can I use sklearn library for implementing chi-square feature selection?

Can you explain what you thinking?

**Comment by parva-jain** [comment]
Actually im thinking of calculating chi-squared score for all features and and then selecting k features with top score. Its similar to PearsonCorrelation class but scoring method is chi-squared this time.

**Comment by sidhu1012** [comment]
Go for it!

**Comment by parva-jain** [comment]
So i can use sklearn library in the script, right?

**Comment by sidhu1012** [comment]
> So i can use sklearn library in the script, right?
> 

Yes I think so

**Comment by parva-jain** [comment]
Hey I'm facing a problem in designing a test script for chi-square feature selection as for cross-checking, I'm not able to get any implementation of it apart from sklearn's which I'm using in the ChiSquare Class.

**Comment by sidhu1012** [comment]
> Hey I'm facing a problem in designing a test script for chi-square feature selection as for cross-checking, I'm not able to get any implementation of it apart from sklearn's which I'm using in the ChiSquare Class.

Start a PR we will figure them out.
